bentoml/yatai/client/bento_repository_api.py

```diff
@@ -200,21 +267,38 @@ def upload_from_dir(self, saved_bento_path, labels=None):
             )
 
         if response.uri.type == BentoUri.LOCAL:
-            if os.path.exists(response.uri.uri):
-                # due to copytree dst must not already exist
-                shutil.rmtree(response.uri.uri)
-            shutil.copytree(saved_bento_path, response.uri.uri)
-
-            self._update_bento_upload_progress(bento_service_metadata)
+            # For remote yatai service, call upload method.
+            if isinstance(self.yatai_service, YataiStub):
+                try:
+                    self.upload_bento(
+                        bento_service_metadata.name,
+                        bento_service_metadata.version,
+                        saved_bento_path,
+                    )
+                    upload_status = UploadStatus.DONE
+                except Exception as e:  # pylint: disable=broad-except
+                    logger.error(e)
+                    upload_status = UploadStatus.ERROR
+            else:
+                if os.path.exists(response.uri.uri):
+                    # due to copytree dst must not already exist
+                    shutil.rmtree(response.uri.uri)
+                shutil.copytree(saved_bento_path, response.uri.uri)
+                upload_status = UploadStatus.DONE
 
-            logger.info(
-                "BentoService bundle '%s:%s' saved to: %s",
-                bento_service_metadata.name,
-                bento_service_metadata.version,
-                response.uri.uri,
+            self._update_bento_upload_progress(
```

Comment:
do we still need this - in the case of calling UploadBento?
UploadBento gRPC should be able to set the upload progress itself once the upload is finished right?

Reply:
I am planning on revisit how bento is added to yatai in a different PR.  One of the change in the upcoming PR will be on progression

Reply:
can we just do that in this PR? it's pretty simple - 1) remove this line here, 2) update the upload status on server side where it handles upload

Reply:
Let's do it in this PR because the logic is wrong here

For local yatai, let's copy the files and call update upload progress to set status to "DONE"

For remote yatai, server-side should update the status upon upload finish